cmd/bb_scheduler/build_queue_state_service.go

```diff
@@ -73,6 +77,44 @@ var (
 		"to_foreground_color": func(s string) string {
 			return "#" + invertColor(s[:2]) + invertColor(s[2:4]) + invertColor(s[4:6])
 		},
+		"to_timestamp": func(t time.Time) string {
+			localLoc, _ := time.LoadLocation("Local")
+			localDateTime := t.In(localLoc)
+			return localDateTime.Format("15:04:05")
+		},
+		"present_if_value": func(format string, s *string) string {
+			if s == nil {
+				return ""
+			}
+			return fmt.Sprintf(format, abbreviateFun(*s))
+		},
+		"list_keywords": func(keyValueMap map[string]string) string {
```

Comment:
Am I correct that the key-value pairs provided by Bazel are never used as if they were a map? It's just an opaque set of values that we get from Bazel that are displayed in the web UI. Instead of letting the build queue state service expose it as a map, I guess it would make more sense to just serve a slice of key-value pairs that are sorted up front?